### PR TITLE
Upgrade slack to v3

### LINKED ISSUE #13515: Update slackapiclient / slack_sdk to v3
Hello,

Slack has released updates to its library and we can start using it for it. 

We especially like one change.

> slack_sdk has no required dependencies. This means aiohttp is no longer automatically resolved.

I've looked through the documentation and it doesn't look like a difficult task, but I think it's still worth testing.
More info: https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x

Best regards,
Kamil Breguła

### PR DESCRIPTION
**Pull request by jyotidhiman0610**
Closes #13515

### CODE REVIEW

**Comment by mik-laj** [comment]
> Importing slack_sdk.* is recommended. You can still use slack.* with deprecation warnings for a while.

https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x

It looks like it will be worth improving the imports.

**Comment by jyotidhiman0610** [comment]
- Taking a look at the failing test.
- Also, verifying _("aiohttp")_ options(might also need some _direction_ here):
> WebClient no longer has run_async and aiohttp specific options. If you still need the option or other aiohttp specific options, use LegacyWebClient (slackclient v2 compatible) or AsyncWebClient

**Comment by mik-laj** [comment]
I pushed a fix for CI problem. "CI Build / Verify Prod Image Py3.6" job was red.

**Comment by jyotidhiman0610** [comment]
Thanks @mik-laj

**Comment by jyotidhiman0610** [comment]
@mik-laj can you please help me with the following:
> WebClient no longer has run_async and aiohttp specific options. If you still need the option or other aiohttp specific options, use LegacyWebClient (slackclient v2 compatible) or AsyncWebClient

What I understand is 
- I need to include `aiohttp` in requirements.txt(conf.py) for it to install as it is not normally resolved now
- To enable `aiohttp.Formdata` option, should I add  `AsyncWebClient` too in the `SlackHook`?
https://github.com/apache/airflow/blob/94d3ed61d60b134d649a4e9785b2d9c2a88cff05/airflow/providers/slack/hooks/slack.py#L97

**Comment by mik-laj** [comment]
@jyotidhiman0610 airflow doesn't support async, so you just need to add an entry in ADDITIONAL_INFO.md. if someone wants to, they can install it additionally, but we don't need to install it automatically, because we don't use it. For example, see: https://github.com/apache/airflow/blob/master/airflow/providers/google/ADDITIONAL_INFO.md

**Comment by jyotidhiman0610** [comment]
@mik-laj added additional information, can you please review the PR?

File: airflow/providers/slack/ADDITIONAL_INFO.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Reference: https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x
+
+### Use  ``AsyncWebClient`` for async options
+
+With v3 upgrade, `WebClient` no longer has `run_async` and `aiohttp` specific options. If you still need the option or other `aiohttp` specific options, use `LegacyWebClient` (slackclient v2 compatible) or `AsyncWebClient`
+
+
+### Include ``aiohttp`` in requirements
+
+`aiohttp` is no longer automatically resolved with `slack_sdk`(earlier `slackclient`). So, if working with `AsyncWebClient`, `AsyncWebhookClient`, and `LegacyWebClient`, it needs separate installation.
```
Comment (mik-laj) [inline airflow/providers/slack/ADDITIONAL_INFO.md]:
```suggestion
## 2.0.0

We updated the support for `sentry_sdk` from ``>=2.0.0,<3.0.0`` to ``>=3.0.0,<4.0.0``. In most cases, this doesn't mean any breaking changes to the DAG files, but if you used this library directly then you have to make the changes. For details, see the `Migration Guide
<https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x>`__ for Python Slack SDK.
```

We should only focus on the changes that affect our project. In this case, there aren't too many breaking changes except that sometimes users may have conflicting dependencies.

File: airflow/providers/slack/ADDITIONAL_INFO.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ specific language governing permissions and limitations
+ under the License.
+ -->
+
+# Migration Guide
+
+## 2.0.0
+
+We updated the support for `slack_sdk` from ``>=2.0.0,<3.0.0`` to ``>=3.0.0,<4.0.0``. In most cases, this doesn't mean any breaking changes to the DAG files, but if you used this library directly then you have to make the changes.
+For details, see the `Migration Guide` (https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x) for Python Slack SDK.
```
Comment (mik-laj) [inline airflow/providers/slack/ADDITIONAL_INFO.md]:
```suggestion
For details, see [the Migration Guide](https://slack.dev/python-slack-sdk/v3-migration/index.html#from-slackclient-2-x) for Python Slack SDK.
```

**Review by mik-laj — APPROVED** [review]

**Pull request merged**